src/simplewallet/simplewallet.cpp

```diff
@@ -8803,22 +8807,36 @@ void simple_wallet::check_for_messages()
 //----------------------------------------------------------------------------------------------------
 void simple_wallet::wallet_idle_thread()
 {
+  const boost::posix_time::ptime start_time = boost::posix_time::microsec_clock::universal_time();
   while (true)
   {
     boost::unique_lock<boost::mutex> lock(m_idle_mutex);
     if (!m_idle_run.load(std::memory_order_relaxed))
       break;
 
+    // if another thread was busy (ie, a foreground refresh thread), we'll end up here at
+    // some random time that's not what we slept for, so we should not call refresh now
+    // or we'll be leaking that fact through timing
+    const boost::posix_time::ptime now0 = boost::posix_time::microsec_clock::universal_time();
+    const auto dt_actual = (now0 - start_time).total_microseconds() % 1000000;
+    if (dt_actual < 1000) // if less than a millisecond... would a very slow machine always miss it ?
```

Comment:
This condition can only trigger once at all because start_time is const and outside the outer loop. Is this intended?

Reply:
I don't see how. Did you consider the modulo ?

Reply:
Ah, so you're trying to always execute this code at the beginning of a second with 1 ms precision. Walking on thin ice I would say, and it definitely needs testing on slow machine like Raspberry Pi.

Reply:
Yes.